### PR TITLE
hw-34

### PR DESCRIPTION
**Pull request by julia-shvabovich**

### CODE REVIEW

**Review by viaviaj — APPROVED** [review]
well done!

File: src/main/java/application/model/User.java [hunk 1]
```diff
@@ -22,7 +26,12 @@ public User(Long id, String name, String login, String password) {
         this.password = password;
     }
 
-    public User() {
+    public User(Long id, String name, String login, String password, byte[] salt) {
```
Comment (KseniiaMakarova) [inline src/main/java/application/model/User.java]: so many constructors, do we use them all?

File: src/main/java/application/util/HashUtil.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        return hashedPassword.toString();
+    }
+
+    public static boolean isValid(String password, User user) {
+        User userFromDB = new User();
+        byte[] salt = user.getSalt();
+        userFromDB.setPassword(hashPassword(password, salt));
+        userFromDB.setSalt(salt);
+        return hashPassword(password, salt).equals(userFromDB.getPassword());
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/application/util/HashUtil.java]: let's make this method private in AuthenticationServiceImpl
also, why are you creating a User instance here?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/application/model/User.java (hunk 1 above)
Comment (julia-shvabovich) [reply src/main/java/application/model/User.java]: done

File: src/main/java/application/util/HashUtil.java (hunk 2 above)
Comment (julia-shvabovich) [reply src/main/java/application/util/HashUtil.java]: done

File: src/main/java/application/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -16,9 +16,14 @@
...
-        if (user.isEmpty() || !HashUtil.isValid(password, user.get())) {
+        if (user.isEmpty() || !isValid(password, user.get())) {
             throw new AuthenticationException("Incorrect username or password");
         }
         return user.get();
     }
+
+    private boolean isValid(String password, User user) {
+        byte[] salt = user.getSalt();
+        return HashUtil.hashPassword(password, salt).equals(HashUtil.hashPassword(password, salt));
```
Comment (KseniiaMakarova) [inline src/main/java/application/security/AuthenticationServiceImpl.java]: ^^ it's a win-win situation) let's reconsider this line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/application/security/AuthenticationServiceImpl.java (hunk 3 above)
Comment (julia-shvabovich) [reply src/main/java/application/security/AuthenticationServiceImpl.java]: done...

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**